client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,279 @@
+				// Go back to general settings if switched to a Jetpack site
+				page( this.getGeneralSettingsUrl( get( nextProps, 'site.slug', '' ) ) );
+				return;
+			}
+
+			// Update state when switching sites
+			this.setState( this.stateForSite( nextProps.site ) );
+		}
+	},
+
+	stateForSite: function( site ) {
+		return {
+			seoMetaDescription: get( site, 'options.seo_meta_description', '' ),
+			verificationServicesCodes: {
+				google: get( site, 'options.verification_services_codes.google', '' ),
+				bing: get( site, 'options.verification_services_codes.bing', '' ),
+				pinterest: get( site, 'options.verification_services_codes.pinterest', '' ),
+				yandex: get( site, 'options.verification_services_codes.yandex', '' )
+			}
+		};
+	},
+
+	handleMetaChange( event ) {
+		this.setState( { seoMetaDescription: get( event, 'target.value', '' ) } );
+	},
+
+	handleVerificationCodeChange( event, serviceName ) {
+		const servicesCodes = this.state.verificationServicesCodes;
+		if ( ! servicesCodes.hasOwnProperty( serviceName ) ) {
+			return;
+		}
+
+		servicesCodes[ serviceName ] = get( event, 'target.value', '' );
+
+		this.setState( { verificationServicesCodes: servicesCodes } );
+	},
+
+	submitSeoForm( event ) {
+		const site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const filteredCodes = omitBy( this.state.verificationServicesCodes, isBoolean );
+		const updatedOptions = {
+			seo_meta_description: this.state.seoMetaDescription,
+			verification_services_codes: filteredCodes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { submittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { submittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+
+		analytics.tracks.recordEvent( 'calypso_seo_settings_form_submit' );
+	},
+
+	getGeneralSettingsUrl( slug ) {
+		return `/settings/general/${ slug }`;
+	},
+
+	render() {
+		const { slug = '', settings: { blog_public = 1 } = {}, fetchingSettings } = this.props.site;
+		const { submittingForm, seoMetaDescription, verificationServicesCodes } = this.state;
+
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
+		const isDisabled = isSitePrivate || fetchingSettings || submittingForm;
+		const isMetaError = seoMetaDescription && seoMetaDescription.length > 160;
+
+		const sitemapUrl = `https://${ slug }/sitemap.xml`;
+		const generalSettingsUrl = this.getGeneralSettingsUrl( slug );
```

Comment:
Also, since this function doesn't depend on the React instance at all, it's a good habit to declare it outside of the React component, up in the module above the `React.createClass()` which will make it a "static" function and slightly more performant (not that it matters here), plus you can then drop `this.`

```js
const getGeneralSettingsUrl = slug => `/settings/general/${ slug }`;

export const SeoForm = React.createClass( {
	…
	const generalSettingsUrl = getGeneralSettingsUrl( slug );
	…
} );
```